test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java

```diff
@@ -0,0 +1,108 @@
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual AWTPanelSmoothWheel
+ */
+
+public class AWTPanelSmoothWheel {
+    private static int wheelEventCount = 0;
+    private static int hiResWheelCount = 0;
+    private static final String INSTRUCTIONS = """
+            <html>
+            <body>
+            This test is relevant on platforms with high-resolution mouse wheel,
+            please press PASS if this is not the case.<br> <br>
+
+            Place the mouse cursor above the green panel and rotate the mouse wheel,
+            the test will print mouse wheel event messages in the format
+            <b> [Event#, WheelRotation, PreciseWheelRotation]</b> into the logging
+            panel below the instruction window.<br> <br>
+
+            A hi-res mouse is one which produces MouseWheelEvents having
+            <b>preciseWheelRotation &lt 1.</b> <br>
+            When preciseWheelRotation adds up to 1,wheelRotation becomes 1. <br>
+            You should see a few events where preciseWheelRotation < 1 & wheelRotation = 0
+            followed by a event where preciseWheelRotation = 1 & wheelRotation = 1.<br> <br>
+
+            Check if the test works OK when the mouse wheel is rotated very slow.<br> <br>
+            Please press PASS if above is true, else FAIL. <br> <br>
+
+            <hr>
+            PS: If you don't see events with preciseWheelRotation < 1,
+            then the mouse doesn't support high-resolution scrolling.
+            A warning is shown in the logging area if you are not using a hi-res mouse.
+            <br> <br>
+            </body>
+            </html>
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows(20)
+                .columns(50)
+                .testTimeOut(10)
+                .logArea(10)
+                .testUI(AWTPanelSmoothWheel::createUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static Frame createUI() {
+        Frame frame = new Frame("Test Wheel Rotation");
+        Panel panel = new Panel();
+        panel.setBackground(Color.GREEN);
+        panel.addMouseWheelListener(e -> {
+            if (e.getScrollType() == MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+                PassFailJFrame.log("WheelEvent#"+ ++wheelEventCount
+                        + " --- Wheel Rotation: " + e.getWheelRotation()
+                        + " --- Precise Wheel Rotation: "
+                        + String.format("%.2f", e.getPreciseWheelRotation()));
+                if (e.getPreciseWheelRotation() < 1) {
+                    hiResWheelCount++;
+                }
+                if (wheelEventCount >= 5 && hiResWheelCount == 0) {
+                    PassFailJFrame.log("WARNING !!! You might not be using a high-res mouse.");
```

Comment:
[`JOptionPane.showMessageDialog`](https://docs.oracle.com/en/java/javase/21/docs/api/java.desktop/javax/swing/JOptionPane.html#showMessageDialog(java.awt.Component,java.lang.Object,java.lang.String,int)) could be a better option to attract the tester's attention.

Reply:
Thanks for the suggestion. Added it.